createAccountArray.js

```diff
@@ -0,0 +1,42 @@
+
+
+class Account {
+    constructor(owner,balancePounds,balancePennies){
+        this.owner=owner;
+        this.balancePounds=balancePounds;
+        this.balancePennies=balancePennies;
+    }
+}
+
+function removeDuplicates(inputArray) {
+    for (let i=0;i<inputArray.length;i++) {
+        const testElement=inputArray[i];
+        for (j=0;j<inputArray.length;j++) {
+            if (inputArray[i]===inputArray[j]&&i!==j) {
+                inputArray.splice(j,1);
+                j--;
+            }
+
+        }
+
+    }
+}
+
+function createAccountArray(allTransactions) {
+    let allFromNames=[];                         
+    let allToNames=[];
+    for (let i=0; i<allTransactions.length;i++) {
+        allFromNames[i]=allTransactions[i].FromAccount;
+        allToNames[i]=allTransactions[i].ToAccount;
+    }

```

Comment:
A nicer way of populating arrays inside a for loop is using `push`, e.g.

    allFromNames.push(allTransactions[i].FromAccount);

However in this case our goal is creating a new array based on the items in our existing array, and there is a method called `map` which does exactly that:

    const allFromNames = allTransactions.map(transaction => transaction.FromAccount);

This does mean that we go over the transaction list twice (once to generate the froms and once to generate the tos), however I'd say it's worth it for the improvement readability. Our code now more closely matches the description of the problem: "allFromNames contains the FromAccount from each transaction".